### PR TITLE
chore: AZCore replace usage of VectorConversions

### PR DESCRIPTION
**Pull request by pollend**
Signed-off-by: Michael Pollind <mpollind@gmail.com>

## What does this PR do?

replace usage of deprecated VectorConversions

### CODE REVIEW

**Review by gadams3 — APPROVED** [review]

**Review by hultonha — APPROVED** [review]

File: Gems/LmbrCentral/Code/Source/Shape/EditorPolygonPrismShapeComponentMode.cpp [hunk 1]
```diff
@@ -27,7 +29,7 @@ namespace LmbrCentral
         AzToolsFramework::MidpointCalculator midpointCalculator;
         for (const AZ::Vector2& vertex : vertexContainer.GetVertices())
         {
-            midpointCalculator.AddPosition(AZ::Vector2ToVector3(vertex, height));
+            midpointCalculator.AddPosition(AZ::Vector3(vertex, height));
```
Comment (pollend) [inline Gems/LmbrCentral/Code/Source/Shape/EditorPolygonPrismShapeComponentMode.cpp:32]: there is a hidden dependency through VertexContainerInterface.h ...

Comment (hultonha) [reply Gems/LmbrCentral/Code/Source/Shape/EditorPolygonPrismShapeComponentMode.cpp:32]: Ah good spot, is this fixed now? Shall I kick off another build?

Comment (hultonha) [reply Gems/LmbrCentral/Code/Source/Shape/EditorPolygonPrismShapeComponentMode.cpp:32]: Nice, PR is green, will merge first thing tomorrow :+1:

Comment (lumberyard-employee-dm) [reply Gems/LmbrCentral/Code/Source/Shape/EditorPolygonPrismShapeComponentMode.cpp:32]: I'll merge it now.

Comment (hultonha) [reply Gems/LmbrCentral/Code/Source/Shape/EditorPolygonPrismShapeComponentMode.cpp:32]: Thank you! 🙂

**Pull request merged**